Bottom: f670071efe52d65e6a223b0f4d819af48505465e
Top:    dfeed2f52eb6f65a9782c15a2e73c0464df3b887
Author: Sean Hefty <sean.hefty@intel.com>
Date:   2012-10-24 15:25:51 -0700

Refresh of riostream

---

diff --git a/examples/riostream.c b/examples/riostream.c
index 35f7eaa..cc7d888 100644
--- a/examples/riostream.c
+++ b/examples/riostream.c
@@ -359,7 +359,7 @@ static int run_test(void)
 			if (ret)
 				goto out;
 
-			ret = recv_xfer(transfer_size, marker++)
+			ret = recv_xfer(transfer_size, marker++);
 		} else {
 			ret = recv_xfer(transfer_size, marker++);
 			if (ret)
@@ -401,7 +401,7 @@ static void set_options(int rs)
 
 	val = 1;
 	rsetsockopt(rs, IPPROTO_TCP, TCP_NODELAY, (void *) &val, sizeof(val));
-	rsetsockopt(rs, SOL_RDMA, RDMA_IOMAP, (void *) &val, sizeof val);
+	rsetsockopt(rs, SOL_RDMA, RDMA_IOMAPSIZE, (void *) &val, sizeof val);
 
 	if (flags & MSG_DONTWAIT)
 		rfcntl(rs, F_SETFL, O_NONBLOCK);
@@ -539,7 +539,7 @@ static int client_connect(void)
 
 close:
 	if (ret)
-		rs_close(rs);
+		rclose(rs);
 free:
 	freeaddrinfo(res);
 	return ret;
@@ -575,14 +575,8 @@ static int run(void)
 			init_latency_test(test_size[i].size);
 			run_test();
 		}
-		if (fork_pid)
-			wait(NULL);
-		else
-			rs_shutdown(rs, SHUT_RDWR);
-		rs_close(rs);
-
-		if (!dst_addr && use_fork && !fork_pid)
-			goto free;
+		rshutdown(rs, SHUT_RDWR);
+		rclose(rs);
 
 		optimization = opt_bandwidth;
 		ret = dst_addr ? client_connect() : server_connect();
@@ -599,15 +593,11 @@ static int run(void)
 		if (ret)
 			goto free;
 
-		if (!fork_pid)
-			ret = run_test();
+		ret = run_test();
 	}
 
-	if (fork_pid)
-		wait(NULL);
-	else
-		rs_shutdown(rs, SHUT_RDWR);
-	rs_close(rs);
+	rshutdown(rs, SHUT_RDWR);
+	rclose(rs);
 free:
 	free(buf);
 	return ret;
